src/System.Text.Encoding.CodePages/tests/EncoderFallbackBufferHelper.cs

```diff
@@ -0,0 +1,88 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using Xunit;
+
+namespace System.Text.Tests
+{
+    public class TestEncoderFallbackBuffer : EncoderFallbackBuffer
+    {
+        private int _fallbackCount = 0;
+
+        public TestEncoderFallbackBuffer() { }
+
+        public override bool Fallback(char charUnknownHigh, char charUnknownLow, int index)
+        {
+            _fallbackCount++;
+            return true;
+        }
+
+        public override bool Fallback(char charUnknown, int index)
+        {
+            _fallbackCount++;
+            return true;
+        }
+
+        public override bool MovePrevious()
+        {
+            if (_fallbackCount > 0)
+            {
+                return true;
+            }
+
+            return false;
+        }
+
+        public override int Remaining { get { return _fallbackCount; } }
+
+        public override char GetNextChar()
+        {
+            if (_fallbackCount > 0)
+            {
+                _fallbackCount--;
+                return '?';
+            }
+            return '\0';
+        }
+    }
+
+    public class TestEncoderFallback : EncoderFallback
+    {
+        public TestEncoderFallback() { }
+
+        public override EncoderFallbackBuffer CreateFallbackBuffer()
+        {
+            return new TestEncoderFallbackBuffer();
+        }
+
+        public override int MaxCharCount { get { return 2; } }
```

Comment:
Nit: could be simplified to `public override int MaxCharCount => 2;`.  Similarly the above could be `public override EncoderFallbackBuffer CreateFallbackBuffer() => new TestEncoderFallbackBuffer();`.